core/src/main/java/tc/oc/pgm/kits/KitParser.java

```diff
@@ -203,7 +203,11 @@ public FlyKit parseFlyKit(Element el) throws InvalidXMLException {
     }
     ItemStack stack = parseItem(el, true);
     boolean locked = XMLUtils.parseBoolean(el.getAttribute("locked"), false);
-    return new ArmorKit.ArmorItem(stack, locked);
+    ArmorKit.ArmorItem armorItem = new ArmorKit.ArmorItem(stack, locked);
+    if (stack.getItemMeta() instanceof LeatherArmorMeta) {
+      armorItem.teamColor = XMLUtils.parseBoolean(el.getAttribute("team-color"), false);
```

Comment:
Pass this as part of the constructor, and keep ArmorItem immutable, making teamColor final.
You can do something along the lines of:
`boolean autoColor = stack.getIteamMeta() instanceof LeatherArmorMeta && XMLUtils.parseBoolean(el.getAttribute("team-color"), false);`

It may be worth considering throwing an XML exception if a map combines team-color attribute with non-leather armor

Reply:
I have added the changes you requested